### DISCUSSION TITLE
Recursively apply namespace under directory

### DISCUSSION THREAD

**Discussion by henrikb77** [fluxcd/flux2#3718] [Q&A]
I'm new to Flux but intend to use it for a new cluster with quite a few apps. Probably the cluster will be sliced with namespaces dev, qa, prod or similar. I will mostly do Helm since it fits well with abstracting stuff from other devs.

I want to split my GitOps repo in a similar fashion:
dev/, qa/, prod/.

Is it possible in some way to make sure everything under e.g. dev/ goes to the dev namespace.

Otherwise people will need to fill out for HelmRelease 'spec.chart.spec.targetNamespace' (or with similar result 'spec.namespace') for every application. And someone WILL copy/paste and keep "dev" under the "prod" directory leading to prod workload ending up in the wrong namespace.

In my ideal scenario I could put a namespace resource in a directory and by applying some config like 'recursive: true' get it to apply to everything beneath that directory.

Being new to Flux it's possible I missed this is possible or there's some other way to get similar results.

**Comment by makkes** [comment]
> Is it possible in some way to make sure everything under e.g. dev/ goes to the dev namespace.

If you use kustomize you can use the `namespace` field: https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/#setting-cross-cutting-fields

Flux Kustomizations have a `.spec.targetNamespace` field but if you have e.g. HelmRelease manifests in there you'd have to make sure they deploy into that namespace as well.

**Comment by henrikb77** [comment]
Thanks for the response.

Even for the Kustomize approach I would have to repeat that for every set of resources (=application) I would like to deploy I guess?

But I think it would be a nice feature to be able to say everything beyond this directory goes into namespace x, or even for root dir - everything in this *repo* goes into namespace x. But given Flux' (as it seems) Kustomize-centric approach it may be hard to accomplish in a clean way.
